### PR TITLE
Add use_blank_slate()

### LINKED ISSUE #133: use_freshstarts()
Re: https://github.com/tidyverse/tidyverse.org/pull/64#discussion_r156173690

Have a function to report or change behaviour around saving/loading `.RData`?

### PR DESCRIPTION
**Pull request by jennybc**
Closes #133.

Implements ~`use_freshstarts()`~ `use_blank_slate()`, to suppress save/reload of user's workspace upon R restarts. 

Handles `scope = "project"` but not `scope = "user"`. Reasons given in https://github.com/r-lib/usethis/issues/133#issuecomment-352918047.

I needed to use work done in PRs #145 (Make scoped_temporary_package() easier to work with) and #146 (Less promiscuous Rbuildignoring) so those need approval first.

### CODE REVIEW

File: R/rstudio.R [hunk 1]
```diff
@@ -19,6 +19,69 @@ use_rstudio <- function() {
...
+#' user interface.
+#'
+#' @inheritParams edit
+#'
+#' @export
+use_freshstarts <- function(scope = c("user", "project")) {
+  scope <- match.arg(scope)
+
+  if (scope == "user") {
+    message(
```
Comment (jennybc) [inline R/rstudio.R]: I'm suspect this appals. Happy to implement in a more usethis-style.

File: R/rstudio.R [hunk 2]
```diff
@@ -19,6 +19,69 @@ use_rstudio <- function() {
...
+
+  if (!is_rstudio_project()) {
+    stop(project_name(), " is not an RStudio Project", call. = FALSE)
+  }
+
+  rproj_options <- build_rproj(
+    rproj_path(),
+    list(RestoreWorkspace = "No", SaveWorkspace = "No")
+  )
+  write_utf8(file.path(proj_get(), rproj_path()), rproj_options)
```
Comment (jennybc) [inline R/rstudio.R]: I would prefer `write_over()` but I don't want to ask user's permission.

File: R/rstudio.R [hunk 3]
```diff
@@ -19,6 +19,69 @@ use_rstudio <- function() {
...
+  restart_rstudio(
+    "A restart of RStudio is required to activate fresh starts"
+  )
```
Comment (jennybc) [inline R/rstudio.R]: I confirmed restart is necessary for this to take effect.

File: R/rstudio.R (hunk 1 above)
Comment (jennybc) [reply R/rstudio.R]: done 0dbc25ac7608d618d98a49722548a10ad5b5ce14

**Comment by hadley** [comment]
Let's call it `use_blank_slate()`

File: R/rstudio.R [hunk 4]
```diff
@@ -32,3 +95,37 @@ in_rstudio <- function(base_path = proj_get()) {
 
   normalizePath(proj) == normalizePath(base_path)
 }
+
+build_rproj <- function(file, fields) {
```
[the remark this answers is not in the document]
Comment (jennybc) [reply R/rstudio.R]: Tests added. This is ready for another look.

[the remark this answers is not in the document]
Comment (jennybc) [reply R/rstudio.R]: I needed to use work done in PRs #145 (Make scoped_temporary_package() easier to work with) and #146 (Less promiscuous Rbuildignoring) so those need approval first.

File: .Rbuildignore [hunk 5]
```diff
@@ -1,4 +1,4 @@
-^.*\.Rproj$
```
Comment (hadley) [inline .Rbuildignore]: Doesn't RStudio modify this when you re-open?

File: tests/testthat/test-rstudio.R [hunk 6]
```diff
@@ -0,0 +1,79 @@
+context("rstudio")
+
+test_that("use_rstudio() creates .Rproj file, named after directory", {
+  dir <- scoped_temporary_package(rstudio = FALSE)
+  capture_output(use_rstudio())
+  rproj <- list.files(proj_get(), pattern = "\\.Rproj$")
+  expect_identical(rproj, paste0(basename(dir), ".Rproj"))
+})
+
+test_that("a non-RStudio project is recognized", {
```
Comment (hadley) [inline tests/testthat/test-rstudio.R]: is _not_ recognised?

File: tests/testthat/test-rstudio.R [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+
+test_that("we can roundtrip an Rproj file", {
+  scoped_temporary_package(rstudio = TRUE)
+  tmp <- tempfile()
+  rproj <- modify_rproj(
+    file.path(proj_get(), rproj_path()),
+    list()
+  )
+  writeLines(serialize_rproj(rproj), tmp)
+  expect_equivalent(
```
Comment (hadley) [inline tests/testthat/test-rstudio.R]: Why not `expect_equal()` on the lines themselves?

File: .Rbuildignore (hunk 5 above)
Comment (jennybc) [reply .Rbuildignore]: It does not for me. Does it for you?

File: tests/testthat/test-rstudio.R (hunk 6 above)
Comment (jennybc) [reply tests/testthat/test-rstudio.R]: ok changed

File: tests/testthat/test-rstudio.R (hunk 7 above)
Comment (jennybc) [reply tests/testthat/test-rstudio.R]: ok changed

**Comment by jennybc** [comment]
@hadley look again?

File: .Rbuildignore (hunk 5 above)
Comment (hadley) [reply .Rbuildignore]: Hmmm, I could've sworn it did in the past, but not this time!

File: R/rstudio.R [hunk 8]
```diff
@@ -13,12 +13,88 @@ use_rstudio <- function() {
 
   use_git_ignore(".Rproj.user")
   if (is_package()) {
-    use_build_ignore(c("^.*\\.Rproj$", "^\\.Rproj\\.user$"), escape = FALSE)
+    use_build_ignore(
+      c(
```
Comment (hadley) [inline R/rstudio.R]: I think you could now make this:

```R
use_build_ignore(c(paste0(project_name(), "\.Rproj"), ".Rproj.user"))
```

File: .Rbuildignore (hunk 5 above)
Comment (jennybc) [reply .Rbuildignore]: I have even checked with a full "new Project (that is a package) from GitHub" workflow. And RStudio doesn't modify this in `.Rbuildignore`

I've come to the conclusion that the problem of the missing template was self-inflicted. Although maybe that `.Rbuildignore` line was originally inspired by RStudio?

Comment (hadley) [reply .Rbuildignore]: Agreed x2

File: R/rstudio.R (hunk 8 above)
Comment (jennybc) [reply R/rstudio.R]: yes, changed

**Pull request merged**